src/Atmos/Model/source.jl

```diff
@@ -173,3 +181,73 @@ function atmos_source!(
         source.ρu -= β_sponge * (state.ρu .- state.ρ * s.u_relaxation)
     end
 end
+
+"""
+    RemovePrecipitation{FT} <: Source
+
+`q_tot` removal when cloud condensate is exceeding a threshold.
+The threshold is defined either in terms of condensate or supersaturation.
+The removal rate is implemented as a relaxation term
+in the Microphysics_0M module.
+The default thresholds and timescale are defined in CLIMAParameters.jl.
+"""
+struct RemovePrecipitation <: Source
+    " Set to true if using qc based threshold"
+    use_qc_thr::Bool
+end
+function atmos_source!(
+    s::RemovePrecipitation,
+    atmos::AtmosModel,
+    source::Vars,
+    state::Vars,
+    diffusive::Vars,
+    aux::Vars,
+    t::Real,
+    direction,
+)
+    # TODO - should I be using aux here? Or do another saturation adjustement?
+    # TODO - how to assert that it is used with EquilMoist module?
+    #        (if it were to be used with non-EquilMoist model
+    #         q_liq and q_ice would be part of state)
+    FT = eltype(state)
+    if aux.moisture.q_liq + aux.moisture.q_ice > eps(FT)
+
+        q = PhasePartition(
+            state.moisture.ρq_tot / state.ρ,
+            aux.moisture.q_liq,
+            aux.moisture.q_ice,
+        )
+        T::FT = aux.moisture.temperature
+        q_dry::FT = 1 - q.tot
+
+        dqt_dt::FT = 0
+        if s.use_qc_thr
+            dqt_dt = remove_precipitation(atmos.param_set, q)
+        else
+            ts_neq = TemperatureSHumNonEquil(atmos.param_set, T, state.ρ, q)
+            q_vap_sat = q_vap_saturation(ts_neq)
+            dqt_dt = remove_precipitation(atmos.param_set, q, q_vap_sat)
+        end
+
+        _e_int_i0::FT = e_int_i0(atmos.param_set)
+        _cv_d::FT = cv_d(atmos.param_set)
+        _cv_l::FT = cv_l(atmos.param_set)
+        _cv_i::FT = cv_i(atmos.param_set)
+        _T_0::FT = T_0(atmos.param_set)
+
+        source.moisture.ρq_tot += state.ρ / q_dry * dqt_dt
+
+        source.ρ += state.ρ / q_dry * dqt_dt
+
+        source.ρe +=
+            (
+                q.liq / (q.liq + q.ice) * (_cv_l - _cv_d) * (T - _T_0) +
+                q.ice / (q.liq + q.ice) *
+                ((_cv_i - _cv_d) * (T - _T_0) - _e_int_i0) +
+                state.ρe / state.ρ / q_dry
+            ) *
+            state.ρ *
+            dqt_dt
```

Comment:
Would it be simpler to just write it as (according to Eq.3.13, also I don't think they are exactly the same):
```
source.ρe +=
            (
                q.liq / (q.liq + q.ice) * _cv_l * (T - _T_0) +
                q.ice / (q.liq + q.ice) *
                (_cv_i * (T - _T_0) - _e_int_i0) +
                gravitational_potential(atmos.orientation, aux)
            ) *
            state.ρ *
            dqt_dt
```

Reply:
I have added the forgotten geopotential. I still think there should also be terms proportional to `-cv_d` there. I have updated the code and the docs. Let me know what you think now.

Reply:
Did you remove the `state.ρe / state.ρ / q_dry` term as well? Now I'm confused by why it was there before.

Reply:
Yes, I did remove it. I think it was there because of my confusion with the notation and derivatives. I think what you suggested is right, modulo the missing `cv_d` terms (?)